pandas/core/resample.py

```diff
@@ -1653,10 +1654,17 @@ def _get_period_bins(self, ax: PeriodIndex):
             nat_count = np.sum(memb._isnan)
             memb = memb[~memb._isnan]
 
-        # if index contains no valid (non-NaT) values, return empty index
         if not len(memb):
-            binner = labels = PeriodIndex(data=[], freq=self.freq, name=ax.name)
-            return binner, [], labels
+            if len(ax) == 0:
+                # if index is empty, return empty bins
+                data = []
+                bins = np.array([], dtype=np.int64)
+            else:
+                # if index is all NaT, return a single bin
+                data = [NaT]
+                bins = np.array([len(ax)])
```

Comment:
wait why do we have NaT in an index? e.g. why don't we have 0 bins?

Reply:
Below this is the logic for when there are `NaT` along with non-`NaT` (lines 1712-1718 here):

```
        if nat_count > 0:
            # NaT handling as in pandas._lib.lib.generate_bins_dt64()
            # shift bins by the number of NaT
            bins += nat_count
            bins = np.insert(bins, 0, nat_count)
            binner = binner.insert(0, NaT)
            labels = labels.insert(0, NaT)
```

The added logic agrees with this in the case there are all NaT.

Reply:
i c. can you unify this logic (maybe to a function)?